### PR TITLE
Implement `gossip.Set#Get`

### PR DESCRIPTION
**Pull request by joshua-kim**
## Why this should be merged

Required as a pre-req to migrate our push-based gossip

## How this works

Implements the `Get` interface. This is used so that we can see if a tx is new or not, to decide on whether we want to forward it to peers.

## How this was tested

Added UTs

### CODE REVIEW

File: plugin/evm/vm.go [hunk 1]
```diff
@@ -1147,6 +1131,31 @@ func (vm *VM) initBlockBuilding() error {
...
+
+	atomicTxGossipHandler = &p2p.ValidatorHandler{
+		ValidatorSet: vm.validators,
+		Handler: &p2p.ThrottlerHandler{
+			Throttler: p2p.NewSlidingWindowThrottler(throttlingPeriod, throttlingLimit),
+			Handler:   atomicTxGossipHandler,
+			Log:       vm.ctx.Log,
+		},
+		Log: vm.ctx.Log,
+	}
```
Comment (ceyonur) [inline plugin/evm/vm.go:1153]: is there an equivalent `New` function for this? If so can we use it (or create if possible)?

File: plugin/evm/vm.go [hunk 2]
```diff
@@ -1071,55 +1071,39 @@ func (vm *VM) initBlockBuilding() error {
...
+		ethTxPool,
+		ethTxGossipHandlerConfig,
+		vm.sdkMetrics,
+	)
 	if err != nil {
 		return err
 	}
-
-	atomicTxGossipHandler = &p2p.ValidatorHandler{
+	ethTxGossipHandler = &p2p.ValidatorHandler{
```
Comment (ceyonur) [inline plugin/evm/vm.go:1093]: these reassignments seems a bit complicated. can we use a different naming for these or try to create a separate `New...` function?

File: plugin/evm/vm.go [hunk 3]
```diff
@@ -1071,55 +1071,39 @@ func (vm *VM) initBlockBuilding() error {
...
 		ValidatorSet: vm.validators,
 		Handler: &p2p.ThrottlerHandler{
+			Handler:   ethTxGossipHandler,
```
Comment (ceyonur) [inline plugin/evm/vm.go:1096]: (from comment above) this looks a bit weird and seems like there is a circular dependency.

File: plugin/evm/vm.go [hunk 4]
```diff
@@ -1071,55 +1071,39 @@ func (vm *VM) initBlockBuilding() error {
...
-		Log: vm.ctx.Log,
-	}
-
-	if err := vm.Network.AddHandler(ethTxGossipProtocol, ethTxGossipHandler); err != nil {
-		return err
+		return fmt.Errorf("failed to create eth tx gossip client: %w", err)
 	}
 
-	atomicTxGossipHandler, err = gossip.NewHandler[*GossipAtomicTx](vm.mempool, atomicTxGossipHandlerConfig, vm.sdkMetrics)
+	ethTxGossipHandler, err = gossip.NewHandler[GossipEthTx, *GossipEthTx](
```
Comment (ceyonur) [inline plugin/evm/vm.go:1083]: we definitely need more comments here as there are currently 4 handler references for eth txs (`NewGossipHandler`, `gossip.NewHandler`, ` &p2p.ValidatorHandler`, `gossip.NewPullGossiper`)

**Review by ceyonur — COMMENTED** [review]
Left few comments. I think it would be much cleaner if we merge this into the `refactor-client` PR as they both change same files. Or we can just keep mempool changes here and move other stuff to `refactor-client` if possible.

**Comment by joshua-kim** [comment]
> Left few comments. I think it would be much cleaner if we merge this into the refactor-client PR as they both change same files. Or we can just keep mempool changes here and move other stuff to refactor-client if possible

I think i was low on sleep when I pushed this out... yeah we should definitely be putting the non-mempool stuff into the other PR.

File: plugin/evm/vm.go (hunk 1 above)
Comment (joshua-kim) [reply plugin/evm/vm.go]: Argh I missed this one. I'll add this as a follow-up... https://github.com/ava-labs/avalanchego/issues/2482

File: plugin/evm/vm.go (hunk 3 above)
Comment (joshua-kim) [reply plugin/evm/vm.go]: Stale diff, I think this is fixed now in the refactor PR.

File: plugin/evm/vm.go (hunk 4 above)
Comment (joshua-kim) [reply plugin/evm/vm.go]: Stale diff, I think this is more verbose in the refactor PR

**Comment by joshua-kim** [comment]
Closing because we are removing `Get` in favor of just using an `error` on `Add` to indicate the tx already exists.

**Pull request closed without merging**